File: BackEnd/Utils.Notifications/FactoryNotifications.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Utils.Notifications
{
    public static class FactoryNotifications
    {

        public enum PushNotificationsSystem { PubNub, RealtimeFramework };

        // Determina qué sistema se utiliza para push notifications.
        private static PushNotificationsSystem _currentNotificationSystem = PushNotificationsSystem.PubNub;

        public static INotifications GetInstance() {
            if (_currentNotificationSystem == PushNotificationsSystem.PubNub)
            {
                return new NotificationsPubNub();
            }
            else if (_currentNotificationSystem == PushNotificationsSystem.RealtimeFramework)
            {
                return new NotificationsRealtime();
            }
            else
            {
                return null;
            }
        }
    }
}

File: FrontEnd/MemsysAndroid/app/src/main/java/pis16/memsysandroid/notifications/factories/FactoryNotifications.java
package pis16.memsysandroid.notifications.factories;

import android.app.Activity;

import pis16.memsysandroid.notifications.NotificationsPubNub;
import pis16.memsysandroid.notifications.NotificationsRealtime;
import pis16.memsysandroid.notifications.interfaces.INotifications;
import pis16.memsysandroid.ui.MainActivity;

/**
 * Created by marccio on 29-Aug-16.
 */
public class FactoryNotifications {



    public enum PushNotificationsSystem { PubNub, RealtimeFramework };

    // Determina qué sistema se utiliza para push notifications.
    private static PushNotificationsSystem _currentNotificationSystem = PushNotificationsSystem.PubNub;

    public static INotifications GetInstance(MainActivity caller) {
        if (_currentNotificationSystem == PushNotificationsSystem.PubNub)
        {
            return new NotificationsPubNub(caller);
        }
        else if (_currentNotificationSystem == PushNotificationsSystem.RealtimeFramework)
        {
            return new NotificationsRealtime(caller);
        }
        else
        {
            return null;
        }
    }

}
